README.md

```diff
@@ -24,30 +24,32 @@ steps:
     uses: actions/setup-java@v4
     with:
       distribution: 'temurin'
-      java-version: '17'
+      java-version: 17
   
   # Install the Firebase Emulator Suite
   - name: Start Firebase Emulator Suite
     uses: firebase-emulator-action@v1
     with:
-      # The version of the Firebase CLI to install, (`npm install -g firebase-tools@${firebase-tools-version}`)
+      # Firebase CLI version to install, (`npm install -g firebase-tools@${firebase-tools-version}`)
       firebase-tools-version: 'latest'
 
-      # A comma separated list of emulators to start.
+      # Comma separated list of emulators to start. This example is the default:
       emulators: 'auth,firestore,functions,storage,database'
 
-      # The project ID to use, defaults to 'test-project'.
+      # Project ID to use, defaults to 'test-project'.
       project-id: 'test-project'
 
-      # The maximum number of retries to attempt before failing the action, defaults to 3.
+      # Maximum attempted retries to before action fails, defaults to 3.
       max-retries: '3'
 
-      # The maximum number of checks to perform before failing the retry, defaults to 60.
+      # Maximum checks to perform before retry fails, defaults to 60.
       max-checks: '60'
 
-      # The time to wait between checks in seconds, defaults to 1.
+      # Seconds to wait between checks, defaults to 1.
       wait-time: '1'
 
-      # The port to check for the emulators, defaults to 9099 (Cloud Firestore). Change this if you are using specific emulators. See https://firebase.google.com/docs/emulator-suite/install_and_configure#port_configuration.
+      # Port to check for emulator startup status, defaults to 9099 (Cloud Firestore).
+      # Change this if you are using specific emulators or non-standard ports.
+      # See https://firebase.google.com/docs/emulator-suite/install_and_configure#port_configuration.
       check-port: '9099'
```

Comment:
Just noticed by reading `action.yml` that this is not true.
The script does not use 9099 by default, it is 8080:

https://github.com/invertase/firebase-emulator-action/blob/8332066498d0b22333a4aa2df58d32ff2b22ac71/action.yaml#L33

```suggestion
      # Port to check for emulator startup status, defaults to 8080 (Cloud Firestore).
      # Change this if you are using specific emulators or non-standard ports.
      # See https://firebase.google.com/docs/emulator-suite/install_and_configure#port_configuration.
      check-port: '8080'
```

A better way to do this - now that I've looked - is to access the hub's `emulators` endpoint and get the list of running emulators - I verified it is dynamic and an emulator only appears once it has actually started. You cannot rely on the count because some emulators start others (functions in particular also starts eventarc and tasks...). 

But you can send that through jq and get a list of the emulators running.

```
curl --silent http://localhost:4400/emulators |jq 'keys[]'
"auth"
"database"
"eventarc"
"firestore"
"functions"
"hub"
"logging"
"tasks"
"ui"
```

So instead of checking firestore (which may or may not be in the list), you can use the input list, split it to the expected emulators, then in the check iterate over the expected list to make sure they are all there

Reply:
> A better way to do this - now that I've looked - is to access the hub's emulators endpoint and get the list of running emulators - I verified it is dynamic and an emulator only appears once it has actually started. You cannot rely on the count because some emulators start others (functions in particular also starts eventarc and tasks...).

When the action runs successfully, it prints out a table with running emulators;
<img width="1440" alt="image" src="https://github.com/user-attachments/assets/413b06ab-e990-4738-b538-344d0d2ebb76">


@mikehardy kindly help me understand, at what point it is necessary to access the hub's emulators endpoint for a the list of running emulators ?

Reply:
@HassanBahati to reduce the need for configuration at all. Right now we propose in the example to check the firestore port but what if they are not running firestore? Then they need to change it, and some will forget, or get it wrong and it's not a terrible thing but there is a little bit of effort involved. If instead the whole "check-port" thing was dynamic by interrogating the hub for running emulators and making sure all the emulators listed in the action were up, there's no need for a specific "check-port" config at all. 

Combined with a verification that at least one emulator was in the list of emulators to start, we provably know that all listed emulators are started, no check-port config needed